fix __post_init__ cause infinite recursion in inheritance (#606)
samuelcolvin/pydantic: HISTORY.rst

History
-------

v0.30 (unreleased)
..................

> * fix ``__post_init__`` cause infinite recursion in inheritance, #606 by @Hanaasagi

Reviewer: ```suggestion
* fix infinite recursion with dataclass inheritance and ``__post_init__``, #606 by @Hanaasagi
```
